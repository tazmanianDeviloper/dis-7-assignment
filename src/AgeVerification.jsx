import {useState} from 'react';

export default function AgeVerification() {
    const [age, setAge] = useState('');
    const [errorMessage, setErrorMessage] = useState('');
    const [welcomeMessageVisible, setWelcomeMessageVisible] = useState(false);

    const checkAge = () => {
        const ageInput = parseInt(age);

        if (isNaN(ageInput) || ageInput < 0) {
            setErrorMessage('Your answer is not acceptable! Please put a new one!');
            setAge('');
        } else if (ageInput > 17) {
            setWelcomeMessageVisible(true);
            setErrorMessage('');
        } else {
            setErrorMessage('You must be at least 18 years old to enter.');
        }
    };

    function handleAgeChange(event){
        setAge(event.target.value);
        setWelcomeMessageVisible(false);
        setErrorMessage('');
    }

    return (
        <div>
            <h2>To enter this website, you need to be at least 18 years old.</h2>
            <label htmlFor="age">How old are you? </label>
            <input id="age" value={age} onChange={handleAgeChange} />
            <button onClick={checkAge}>Check Age</button>
            {errorMessage && <h5>{errorMessage}</h5>}
            {welcomeMessageVisible && <h1>Welcome</h1>}
        </div>
    );
}

