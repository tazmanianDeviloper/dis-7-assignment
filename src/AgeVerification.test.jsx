import AgeVerification from './AgeVerification';
import renderer from 'react-test-renderer';
import { test, expect } from 'vitest';

test('The AgeVerification component should render', () => {
    const view = renderer
        .create(<AgeVerification/>)
        .toJSON();
    expect(view).toMatchSnapshot();
});

test('Entering invalid age should display error message', () => {
    const component = renderer.create(<AgeVerification />);
    const ageInput = component.root.findByProps({ id: 'age' });

    // Change input value to an invalid age
    ageInput.props.onChange({ target: { value: 'abc' } });

    // Find button and simulate click
    component.root.findByType('button').props.onClick();

    // Find error message
    const errorMessage = component.root.findByType('h5');

    // Assert error message is displayed
    expect(errorMessage).toBeDefined();
});

test('Entering valid age should display welcome message', () => {
    const component = renderer.create(<AgeVerification />);
    const ageInput = component.root.findByProps({ id: 'age' });

    // Change input value to a valid age
    ageInput.props.onChange({ target: { value: '20' } });

    // Find button and simulate click
    component.root.findByType('button').props.onClick();

    // Find welcome message
    const welcomeMessage = component.root.findByType('h1');

    // Assert welcome message is displayed
    expect(welcomeMessage).toBeDefined();
});

test('Entering age less than 18 should display error message', () => {
    const component = renderer.create(<AgeVerification />);
    const ageInput = component.root.findByProps({ id: 'age' });

    // Change input value to an age less than 18
    ageInput.props.onChange({ target: { value: '15' } });

    // Find button and simulate click
    component.root.findByType('button').props.onClick();

    // Find error message
    const errorMessage = component.root.findByType('h5');

    // Assert error message is displayed
    expect(errorMessage).toBeDefined();
});

test('Entering a decimal age above 17 should display an error message', () => {
    const component = renderer.create(<AgeVerification />);
    const ageInput = component.root.findByProps({ id: 'age' });

    // Change input value to a decimal age above 17
    ageInput.props.onChange({ target: { value: '17.5' } });

    // Find button and simulate click
    component.root.findByType('button').props.onClick();

    // Find error message
    const errorMessage = component.root.findByType('h5');

    // Assert error message is displayed
    expect(errorMessage).toBeDefined();
});

test('Entering an age above 120 should display an error message', () => {
    const component = renderer.create(<AgeVerification />);
    const ageInput = component.root.findByProps({ id: 'age' });

    // Change input value to an age above 120
    ageInput.props.onChange({ target: { value: '121' } });

    // Find button and simulate click
    component.root.findByType('button').props.onClick();

    // Find error message
    const errorMessage = component.root.findByType('h5');

    // Assert error message is displayed
    expect(errorMessage).toBeDefined();
});
