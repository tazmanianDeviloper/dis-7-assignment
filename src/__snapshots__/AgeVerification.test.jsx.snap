// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The AgeVerification component should render 1`] = `
<div>
  <h2>
    To enter this website, you need to be at least 18 years old.
  </h2>
  <label
    htmlFor="age"
  >
    How old are you? 
  </label>
  <input
    id="age"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Check Age
  </button>
</div>
`;
